components/common-go/baseserver/server.go

```diff
@@ -65,25 +96,34 @@ type Server struct {
 
 func (s *Server) ListenAndServe() error {
 	var err error
-	s.grpcListener, err = net.Listen("tcp", fmt.Sprintf(":%d", s.cfg.grpcPort))
+
+	s.debugListener, err = net.Listen("tcp", fmt.Sprintf(":%d", s.cfg.debugPort))
 	if err != nil {
-		return fmt.Errorf("failed to acquire port %d", s.cfg.grpcPort)
+		return fmt.Errorf("failed to acquire port %d", s.cfg.debugPort)
 	}
 
-	s.httpListener, err = net.Listen("tcp", fmt.Sprintf(":%d", s.cfg.httpPort))
-	if err != nil {
-		return fmt.Errorf("failed to acquire port %d", s.cfg.httpPort)
+	if s.isGRPCEnabled() {
+		s.grpcListener, err = net.Listen("tcp", fmt.Sprintf(":%d", s.cfg.grpcPort))
+		if err != nil {
+			return fmt.Errorf("failed to acquire port %d", s.cfg.grpcPort)
+		}
+	}
+
+	if s.isHTTPEnabled() {
+		s.httpListener, err = net.Listen("tcp", fmt.Sprintf(":%d", s.cfg.httpPort))
+		if err != nil {
+			return fmt.Errorf("failed to acquire port %d", s.cfg.httpPort)
+		}
 	}
 
 	errors := make(chan error)
 	defer close(errors)
 	s.listening = make(chan struct{})
 
+	// Always start the debug server, we should always have metrics and other debug information.
 	go func() {
-		s.Logger().
-			WithField("protocol", "grpc").
-			Infof("Serving gRPC on %s", s.grpcListener.Addr().String())
-		if serveErr := s.grpc.Serve(s.grpcListener); serveErr != nil {
+		s.Logger().WithField("protocol", "http").Infof("Serving debug server on %s", s.debugListener.Addr().String())
+		if serveErr := s.debug.Serve(s.debugListener); serveErr != nil {
```

Comment:
I guess, for consistency with the style guide, this `serveErr` assignment should be moved to the line above.